### PR TITLE
faking livewire entangle.lazy

### PR DESCRIPTION
**Pull request by tanthammar**
This is a base for faking `$wire.entangle(...).lazy `.
A 2nd argument `$except`, has been added to `applyStateBindingModifiers()`. (Suggested by Dan)

The `TextInput` has been updated to support `lazy()` on fields using `TextInput\Mask`.

It will not be as easy as adding an `x-on:blur` to all other entangled alpine fields, but this is a start.

### CODE REVIEW

File: packages/forms/resources/views/components/text-input.blade.php [hunk 1]
```diff
@@ -42,10 +42,11 @@
                 @else
                     x-data="textInputFormComponent({
                         {{ $hasMask() ? "getMaskOptionsUsing: (IMask) => ({$getJsonMaskConfiguration()})," : null }}
-                        state: $wire.{{ $applyStateBindingModifiers('entangle(\'' . $getStatePath() . '\')') }},
+                        state: $wire.{{ $applyStateBindingModifiers('entangle(\'' . $getStatePath() . '\')', except: 'lazy') }},
                     })"
                     type="text"
                     wire:ignore
+                    @if (in_array('lazy', $getStateBindingModifiers())) x-on:blur="$wire.$refresh" @endif
```
Comment (danharrin) [inline packages/forms/resources/views/components/text-input.blade.php:49]: We could maybe extract this into a `hasStateBindingModifier('lazy')` method?

File: packages/forms/src/Concerns/HasStateBindingModifiers.php [hunk 2]
```diff
@@ -29,11 +29,17 @@ public function stateBindingModifiers(array $modifiers): static
         return $this;
     }
 
-    public function applyStateBindingModifiers($expression): string
+    public function applyStateBindingModifiers($expression, ?string $except = null): string
     {
         $modifiers = $this->getStateBindingModifiers();
 
-        return implode('.', array_merge([$expression], $modifiers));
+        $merged = array_merge([$expression], $modifiers);
```
Comment (danharrin) [inline packages/forms/src/Concerns/HasStateBindingModifiers.php:36]: Logically, it makes more sense to filter the array to remove the modifiers, then merge the expression

We can probably do it with a collection, chaining `->implode()` at the end

**Review by danharrin — CHANGES_REQUESTED** [review]

File: packages/forms/resources/views/components/text-input.blade.php (hunk 1 above)
Comment (tanthammar) [reply packages/forms/resources/views/components/text-input.blade.php:49]: I it is going to be a separate method, I like short names ;) `@if(isLazy()) ... @endif`

**Comment by tanthammar** [comment]
@danharrin 
I filtered the modifiers.

File: packages/forms/src/Concerns/HasStateBindingModifiers.php (hunk 2 above)
Comment (tanthammar) [reply packages/forms/src/Concerns/HasStateBindingModifiers.php]: I changed it, see comment below.

**Comment by danharrin** [comment]
Thanks!

**Pull request merged**